### PR TITLE
Convert docker and dockerd commands to spf13/cobra

### LINKED ISSUE #23211: Convert commands to spf13/cobra
This is a checklist for tracking which commands are being worked on or are complete:

* [x] attach - #23274 @WeiZhang555 
* [x] build - https://github.com/docker/docker/pull/23343 @vdemeester
* [x] commit - https://github.com/docker/docker/pull/23309 @yongtang
* [x] cp - https://github.com/docker/docker/pull/23348 @vdemeester
* [x] create - #23159 / https://github.com/docker/docker/issues/23253 @dnephin 
* [x] diff- https://github.com/docker/docker/pull/23283 @yongtang
* [x] events - https://github.com/docker/docker/pull/23334 @WeiZhang555 
* [x] exec #23759 @AkihiroSuda  
* [x] export - #23267 
* [x] history - https://github.com/docker/docker/pull/23291 @yongtang
* [x] images - https://github.com/docker/docker/pull/23292 @yongtang
* [x] import - #23269 @vdemeester
* [x] info #23399 @tkopczynski
* [x] inspect #23902 @dnephin
* [x] kill - https://github.com/docker/docker/issues/23297 @WeiZhang555 
* [x] load - #23412 @vdemeester 
* [x] login - #23265
* [x] logout - #23265
* [x] logs - https://github.com/docker/docker/pull/23284 @yongtang 
* [x] network - #23210
* [x] pause - https://github.com/docker/docker/pull/23285 @yongtang
* [x] port - https://github.com/docker/docker/issues/23286 @yongtang
* [x] ps - https://github.com/docker/docker/pull/23373 @capkurmagati
* [x] pull #23256
* [x] push #23256
* [x] rename - https://github.com/docker/docker/pull/23290 @yongtang
* [x] restart - https://github.com/docker/docker/issues/23307 @WeiZhang555 
* [x] rm - https://github.com/docker/docker/pull/23311 @WeiZhang555 
* [x] rmi - #23242
* [x] run - #23159
* [x] save - #23412 @vdemeester
* [x] search - #23241
* [x] start - #23268
* [x] stats - https://github.com/docker/docker/pull/23334 @WeiZhang555 
* [x] stop - #23276 @yongtang 
* [x] tag - #23278 @yongtang 
* [x] top - https://github.com/docker/docker/pull/23306 @yongtang
* [x] unpause - #23279 @yongtang
* [x] update - https://github.com/docker/docker/pull/23281  @yongtang
* [x] version - https://github.com/docker/docker/pull/23333 @yongtang
* [x] volume - #22769
* [x] wait - https://github.com/docker/docker/pull/23287 @yongtang
* [ ] daemon / dockerd #25354 @dnephin

### LINKED ISSUE #25547: daemon config validation does not detect conflicts with "common flags"
I discovered this while working on #25354

`findConfigurationConflicts()` only receives the daemon specific flags, not the "common flags" (debug, log-level, tls*, etc), so if a daemon is started with any of these flags, a config reload could override the previous value without the user being aware.

### PR DESCRIPTION
**Pull request by dnephin**
Fixes #23211 
Fixes #25547 
- removes `pkg/mflag`
- removes most of `cli/` and consolidates the rest of the package into fewer files

### CODE REVIEW

**Comment by tonistiigi** [comment]
@dnephin Test failures seem related.

**Comment by dnephin** [comment]
I think they are related. Having some trouble finding the cause, but still working on it.

**Comment by dnephin** [comment]
This is green!

**Comment by vdemeester** [comment]
@dnephin given the uses of `pkg/mflag` (https://github.com/search?q=github.com%2Fdocker%2Fdocker%2Fpkg%2Fmflag&type=Code&utf8=%E2%9C%93), I don't think we should remove it. Just not using it anymore.

**Comment by dnephin** [comment]
Hmm, I don't think it should be maintained in this repo if it's not used here.

We could put it up at `github.com/docker/mflag` with a README saying that it is no longer maintained.

**Comment by vdemeester** [comment]
@dnephin maybe, but we should, somehow let people know it's moving into a new package 👼

**Comment by dnephin** [comment]
We could put a README in `./pkg/mflag` in this repo to inform users of the package that it has moved to a new location.

**Comment by dnephin** [comment]
> but we should, somehow let people know it's moving into a new package 

They'll find out pretty quickly when the files don't exist anymore :)  I asked around a bit and others seem to agree that we shouldn't keep it here if it's not in use. If we're not going to maintain it, there probably isn't much value in putting it up at another location either.

File: cmd/docker/daemon_none.go [hunk 1]
```diff
@@ -4,12 +4,22 @@ package main
 
 import (
 	"fmt"
+	"github.com/spf13/cobra"
```
Comment (vdemeester) [inline cmd/docker/daemon_none.go]: nit: this import is misplaced 👼

**Comment by vdemeester** [comment]
LGTM 🐸

**Comment by 2opremio** [comment]
> They'll find out pretty quickly when the files don't exist anymore

You bet! We just did.

**Comment by runcom** [comment]
This is breaking master, with this ExecStart (note using the compat `docker daemon`):
```
ExecStart=/usr/bin/docker daemon \
          --exec-opt native.cgroupdriver=systemd \
          $OPTIONS \
          $DOCKER_STORAGE_OPTIONS \
          $DOCKER_NETWORK_OPTIONS \
          $INSECURE_REGISTRY
```
I'm getting:
```
Aug 26 14:49:10 localhost.localdomain docker[25774]: unknown flag: --exec-opt
Aug 26 14:49:10 localhost.localdomain docker[25774]: See 'docker daemon --help'.
```
This has somehow broke the old `docker daemon` compat command

@dnephin

**Comment by runcom** [comment]
@dnephin can we also add a test which uses `docker daemon` to start up a daemon to not regress on this in the future? (at least till we support `docker daemon`)

**Comment by dnephin** [comment]
@runcom ok, i'll take a look, and add the missing test

**Pull request merged**